code/controllers/subsystem/starmap.dm

```diff
@@ -149,6 +175,12 @@ SUBSYSTEM_DEF(starmap)
 
 		from_time = 0
 		to_time = 0
+
+		if(current_planet.visited != TRUE)
+			current_planet.event.activate_event()
```

Comment:
Currently activate_event is called immediately after init_event with no delay. I'm not sure if you intend that.

Regardless, move this conditional to above with the other conditionals. Also the way it currently is, if a planet is returned to it will call init_event again on the event even if activate_event isn't.